server/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -756,7 +756,6 @@ public synchronized void updateMetadata(final IndexMetadata currentIndexMetadata
             if (currentSettingsVersion == newSettingsVersion) {
                 assert updateIndexSettings == false;
             } else {
-                assert updateIndexSettings;
```

Comment:
This assertion _almost_ holds but fails in cases where you set `index.number_of_replicas` and then auto-expand replicas sets it back again. I think it's ok to drop it, it's not really a big deal to process the occasional no-op update.